src/library/dao/playlistdao.cpp

```diff
@@ -414,6 +414,37 @@ void PlaylistDAO::removeHiddenTracks(const int playlistId) {
     emit(changed(playlistId));
 }
 
+void PlaylistDAO::removeEmptyPlaylists(const int hidden) {
+    ScopedTransaction transaction(m_database);
```

Comment:
We don't really need a transaction here for safety reasons, because each playlist can be deleted independently. It also doesn't work as expected, because SQLite doesn't allow nested transactions and deletePlaylist() internally tries to open another transaction that is committed immediately!

For the future: Transactions should be managed by a controller that coordinates operations on multiple DAOs. If operations in the DAO require to be executed within a transaction (because multiple, dependent SQL commands are executed subsequently) then a transaction should be passed as an additional parameter and the operation should check that this transaction is active before performing any SQL commands.

Reply:
Ups. I noticed that some operations in CrateStorage would actually need such a transaction parameter, because multiple mutating SQL commands are executed in a row. We should fix that, after the work of @gramanas on nested crates has been merged.